great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -449,6 +440,128 @@ class MetricDomainTypes.
+                    column_B_name = domain_kwargs["column_B"]
+
+                ignore_row_if = domain_kwargs["ignore_row_if"]
+                if ignore_row_if == "both_values_are_missing":
+                    selectable = (
+                        sa.select([sa.text("*")])
+                        .select_from(selectable)
+                        .where(
+                            sa.and_(
+                                sa.column(column_A_name) != None,
+                                sa.column(column_A_name) != None,
+                            )
+                        )
+                    )
+                elif ignore_row_if == "either_value_is_missing":
+                    selectable = (
+                        sa.select([sa.text("*")])
+                        .select_from(selectable)
+                        .where(
+                            sa.or_(
+                                sa.column(column_A_name) != None,
+                                sa.column(column_A_name) != None,
+                            )
+                        )
+                    )
+                else:
+                    if ignore_row_if != "never":
+                        raise ValueError(
+                            f'Unrecognized value of ignore_row_if ("{ignore_row_if}").'
+                        )
+
+                return selectable
+
+        if "column_list" in domain_kwargs:
+            if "ignore_row_if" in domain_kwargs:
+                if self.active_batch_data.use_quoted_name:
+                    # If case matters...
+                    column_list = [
+                        quoted_name(domain_kwargs[column_name])
+                        for column_name in domain_kwargs["column_list"]
+                    ]
+                else:
+                    column_list = domain_kwargs["column_list"]
+
+                ignore_row_if = domain_kwargs["ignore_row_if"]
+                if ignore_row_if == "all_values_are_missing":
+                    selectable = (
+                        sa.select([sa.text("*")])
+                        .select_from(selectable)
+                        .where(
+                            sa.and_(
+                                *(
+                                    sa.column(column_name) != None
+                                    for column_name in column_list
+                                )
+                            )
+                        )
+                    )
+                elif ignore_row_if == "any_value_is_missing":
+                    selectable = (
+                        sa.select([sa.text("*")])
+                        .select_from(selectable)
+                        .where(
+                            sa.or_(
+                                *(
+                                    sa.column(column_name) != None
+                                    for column_name in column_list
+                                )
+                            )
```

Comment:
I think this is a logic error: it says *keep* if any value is *not* missing, but should say *ignore* if any value *is* missing.

Reply:
@jcampbell You are right. I corrected the logic and added a test case to exercise it. Thanks for catching this.